shared/src/utils/random.ts

```diff
@@ -57,6 +66,35 @@ export default class Random {
     return Math.floor(this.float(min, max + 1))
   }
 
+  /**
+   * Returns a random integer (normal distribution)
+   *
+   * The function tries to generate a number within the range of min and max.
+   * If it fails to do so after 10 tries, it returns the mean value.
+   *
+   * So not perfect, because usually normal distribution does not have a min and max.
+   *
+   * @param min - The minimum number
+   * @param max - The maximum number
+   * @param mean - The expected value (average)
+   * @param stdev - The standard deviation
+   */
+  intNormal(min: number, max: number, mean = Math.round((min + max) / 2), stdev = 1): number {
+    let value = 0
+    // try 10 times to generate a value within min and max
+    // so the probability of failing gets smaller
+    // otherwise return either min or max
+    for (let i = 0; i < 10; i++) {
+      const normal = this.normal()
+      value = Math.round(normal * stdev + mean)
+      if (value >= min && value <= max) {
+        return value
+      }
+    }
+    // return min if value close to min, max if value close to max
+    return Math.abs(value - min) < Math.abs(value - max) ? min : max
```

Comment:
The comment (line 73) says in this case `mean` is returned. So either the comment or the code is wrong. Also, the name of the function suggests that it should return integers, but min and max could be any number. Should this be checked?

Reply:
Yes, I think I talked with Holger about this and changed the code to return either `min` or `max` in case of a failure. But probably forgot to update the docstring. I changed the doc 👍